Lex/Yacc for wSQLx

N00781354 opened the pull request.

nickJones67 committed:
Grammar (so far)

Not quite done with the grammar yet but trying to finish it tonight. Let me know how it looks and if I missed anything/f'd up

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/origin/patch-1'

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into patch-1

nickJones67 committed:
Update and rename Grammar_v2 to Grammar_v3

nickJones67 committed:
Create dumbTimeStamp.c

This creates a timestamp from the system time. Looking at time.h for a better alternative

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/JonathanFrias/master'

Conflicts:
	Steves-bad-grammar

nickJones67 committed:
Merge branch 'master' of https://github.com/nickJones67/cop4710-sequel

nickJones67 committed:
Merge branch 'refs/heads/patch-1'

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Grammar_work

nickJones67 committed:
Update Grammar_v3

no expressions. just literals

nickJones67 committed:
FinalGrammar

This should be the final grammar

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/origin/Grammar_work'

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Grammar_work

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/origin/Grammar_work'

nickJones67 committed:
small grammar update

added some necessary tokens to the list

nickJones67 committed:
Create LexicalAnalyzer.c

lexical analyzer and command prompt user interface for xSQLx engine

nickJones67 committed:
update to timeStamp.c

uses time.h instead

N00781354 committed:
Lex/Yacc for wSQLx

JonathanFrias commented:
:+1: As far as I understand, this will just determine if the user entered a valid command or not! I'm not an expert with lex/yacc, but maybe since you got this working, there is a way to just output tokens?

SCostello84 commented:
I suggest saving the tokens to an arraylist. Lex/yacc is confirmation that
it is a valid string
On Nov 21, 2015 3:40 PM, "Jonathan Frias" notifications@github.com wrote:

> [image: :+1:] As far as I understand, this will just determine if the
> user entered a valid command or not! I'm not an expert with lex/yacc, but
> maybe since you got this working, there is a way to just output tokens?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/JonathanFrias/cop4710-sequel/pull/12#issuecomment-158679557
> .

JonathanFrias commented:
@SCostello84 Yeah, I had a thought based on your idea.

We can alloc a `ParseTree` and fill in the values for it during the actual parse.

And it would do parsing all in one go easier and simpler, and return our struct.

like inside the .l file you should be able to do something like

`parsetree->commandType = SELECT`
`parsetree->(fields+offset) = field`
...
etc.

nickJones67 committed:
minor update to Grammar_v3

added DATE to field type along with the associated grammar rules

nickJones67 committed:
Update timeStamp.c

nickJones67 committed:
Update Grammar_v3

nickJones67 committed:
Update Grammar_v3

nickJones67 committed:
Update Grammar_v3

nickJones67 committed:
Update timeStamp.c

nickJones67 committed:
Create sql.l

nickJones67 committed:
Create sql.y

JonathanFrias commented:
Any progress on this?

nickJones67 commented:
Yea I've made a lot of progress. I haven't been able to upload it yet but I will soon

nickJones67 commented:
LEX is still a little buggy but BISON is working well

nickJones67 committed:
Update sql.l

nickJones67 committed:
Update sql.y

nickJones67 committed:
create makefile

this is the makefile for the sql.l and sql.y only. The executable is called 'trial'

nickJones67 committed:
Rename makefile to makefile for sql.l and sql.y

nickJones67 committed:
Update sql.y

nickJones67 committed:
Update sql.l

nickJones67 committed:
Update sql.y

nickJones67 committed:
Create sql.c

nickJones67 committed:
Update makefile for sql.l and sql.y

nickJones67 committed:
Update sql.l

nickJones67 committed:
Update sql.y

nickJones67 committed:
Update Grammar_v3

nickJones67 committed:
Update sql.y

N00781354 commented:
I think I've got this working now!
I can successfully call yyparse() in a separate C file now. As a heads up, yyparse() has something along the lines of scanf() functionality once called. Currently I just have things set up so that only one sql query is prompted for and processed, but we can change this. (Since we want to allow multiple queries in one entry, right?)  

Along with that, I've successfully made function calls from within the Yacc file using its tokens. (Strings, Integers, etc.) 

All I have to do now is:  
1. Finish up the rest of the grammar cases in the Yacc file; Currently only CREATE_DATABASE_COMMAND makes any function calls and prints its ID. (It's the one I was testing)
2. Edit my Lex file a bit to accept any string case (not just upper) and maybe edit my integers and floats?
3. Properly integrate the Lex and Yacc files into our REAL C program, and make the proper REAL calls to Jonathan's "command" function.

This should be pretty straight-forward based on my examples I'll be uploading shortly. I have to call it quits for now, but this should be done by tomorrow night.

Take a look at how it all works, and if possible, go ahead and make the proper modifications in order to get things rolling. It's not too complicated, IMO, but let me know if any of it seems confusing.

Have a great one guys!

N00781354 committed:
Updated Lex/Yacc functionality

N00781354 committed:
Updated Lex/Yacc functionality

JonathanFrias commented:
:+1:

nickJones67 commented:
Just tested this biz and it seems to be working pretty well! Working on adding your changes to my stuff and make some positive changes. Duplicate and elaborate. It's a walk-off.

nickJones67 committed:
Update common.h

nickJones67 committed:
Update sql.l

nickJones67 committed:
Update sql.y

nickJones67 committed:
Update common.h

nickJones67 committed:
Update common.h

nickJones67 committed:
Update sql.c

Commit:
merge code from upstream

nickJones67 committed:
Merge pull request #2 from JonathanFrias/nickJones67-master

Code Sync with Jonathan

nickJones67 committed:
Update common.h

nickJones67 committed:
Update sql.c

nickJones67 committed:
WIP(Work in Progress) commit

nickJones67 committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	.gitignore
	timeStamp.c

nickJones67 committed:
Do merge

nickJones67 committed:
Merge remote-tracking branch 'refs/remotes/JonathanFrias/master'

nickJones67 committed:
add spaces

nickJones67 committed:
WIP Working on getting flex/bison to work properly

nickJones67 committed:
got stuff to compile

N00781354 committed:
Merge branch 'master' of github.com:nickJones67/cop4710-sequel

N00781354 committed:
Adding SQL Functionality
